sdk/cosmosdb/cosmos/src/client/ChangeFeed/ChangeFeedForPartitionKey.ts

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { ChangeFeedIteratorOptions } from "./ChangeFeedIteratorOptions";
+import { ChangeFeedIteratorResponse } from "./ChangeFeedIteratorResponse";
+import { Resource } from "../../client";
+import { ClientContext } from "../../ClientContext";
+import { Constants, ResourceType } from "../../common";
+import { FeedOptions, Response, ErrorResponse } from "../../request";
+import { ContinuationTokenForPartitionKey } from "./ContinuationTokenForPartitionKey";
+import { ChangeFeedIteratorV2 } from "./ChangeFeedIteratorV2";
+import { PartitionKey } from "../../documents";
+import { PartitionKeyRange } from "../../client";
+import { IEpkRange } from "./IEpkRange";
+
+/**
+ * Provides iterator for change feed.
+ *
+ * Use `Items.getChangeFeedIterator()` to get an instance of the iterator.
+ */
+export class ChangeFeedForPartitionKey<T> extends ChangeFeedIteratorV2<T> {
+  private continuationToken: ContinuationTokenForPartitionKey;
+  private startTime: string;
+
+  /**
+   * @internal
+   */
+  constructor(
+    private clientContext: ClientContext,
+    private resourceId: string,
+    private resourceLink: string,
+    private rId: string,
+    private partitionKey: PartitionKey,
+    private changeFeedOptions: ChangeFeedIteratorOptions
+  ) {
+    super();
+    let canUseStartFromBeginning = true;
+    if (changeFeedOptions.continuationToken) {
+      canUseStartFromBeginning = false;
+      this.continuationToken = JSON.parse(changeFeedOptions.continuationToken);
+    } else {
+      this.continuationToken = new ContinuationTokenForPartitionKey(rId, partitionKey, "");
+    }
+
+    if (changeFeedOptions.startTime) {
+      this.startTime = changeFeedOptions.startTime.toUTCString();
```

Comment:
can we associate time unit with this field something like startTImeInMs

Reply:
No, the expected input is of type Date. 
Eg. startTime: new Date("04/12/2024 07:00:15")